pkg/volume/util/operationexecutor/operation_executor.go

```diff
@@ -389,11 +389,10 @@ func (oe *operationExecutor) DetachVolume(
 }
 
 func (oe *operationExecutor) VerifyVolumesAreAttached(
-	attachedVolumes []AttachedVolume,
-	nodeName types.NodeName,
+	attachedVolumesByNode map[types.NodeName][]AttachedVolume,
 	actualStateOfWorld ActualStateOfWorldAttacherUpdater) error {
 	volumesAreAttachedFunc, err :=
-		oe.operationGenerator.GenerateVolumesAreAttachedFunc(attachedVolumes, nodeName, actualStateOfWorld)
+		oe.operationGenerator.GenerateVolumesAreAttachedFunc(attachedVolumesByNode, actualStateOfWorld)
 	if err != nil {
```

Comment:
The problem is caused by oe.pendingOperations.Run("" /* volumeName */, "" /* podName */, volumesAreAttachedFunc). Because we use empty volumeName, pendingOperation will not check whether there is an existing operation which might check the same set of volumes. I am thinking we might use a string consisting of all the volume names to prevent redundant checking.